lib/puppet/util/json.rb

```diff
@@ -0,0 +1,57 @@
+module Puppet::Util
+  module Json
+    class ParseError < StandardError
+    end
+
+    begin
+      require 'multi_json'
+      # Force backend detection before attempting to use the library
+      # or load any other JSON libraries
+      MultiJson.default_adapter
+
+      # Preserve core type monkey-patching done by the built-in JSON gem
```

Comment:
Is it overwrought to do the monkey patching ourselves and issue a deprecation warning? Is it worth deprecating and removing or would we just end up maintaining the monkey patches ourselves for years?

Reply:
I feel safe just requiring this gem here to bring in the monkey patches because it is shipped with Ruby. We can rely on them keeping the monkey-patches up-to-date (e.g. if new core types are added), and if they deprecate that, we will get the deprecation for free. This feels significantly cleaner to me than trying to maintain our own patches.